battery.go

```diff
@@ -154,3 +155,32 @@ func getAll(sg func() ([]*Battery, error)) ([]*Battery, error) {
 func GetAll() ([]*Battery, error) {
 	return getAll(systemGetAll)
 }
+
+func summarize(sg func() ([]*Battery, error)) (*Battery, error) {
+	bs, err := sg()
+	if err != nil {
+		return nil, err
+	}
+	if len(bs) == 1 {
+		return bs[0], nil
+	}
+	res := new(Battery)
+	for _, battery := range bs {
+		res.Current += battery.Current
+		res.Full += battery.Full
+		res.ChargeRate += battery.ChargeRate
+		res.Design += battery.Design
+		res.Voltage = math.Max(res.Voltage, battery.Voltage)
+		res.DesignVoltage = math.Max(res.DesignVoltage, battery.DesignVoltage)
+		res.State = State(math.Max(float64(res.State), float64(battery.State)))
```

Comment:
Given how the states are ordered, I think it actually makes sense. Sure, there are weird cases like 1 `charging` 1 `discharging`, but in such scenario I'd rather know about the discharging part, I think.

Maybe it would be worth introducing specific states for summarize? Not sure ATM.

What I think would be good is have some more test cases for this, e.g. that for `Full` and `Discharging` it actually returns `Discharging`, etc.